src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -244,22 +244,12 @@ HRESULT CProfilerManager::AddRawProfilerHook(
 
     m_profilerCallbackHolder = std::move(profilerCallbackHolder);
 
-    if (m_attachedClrVersion != ClrVersion_2)
-    {
-        hr = pCorProfilerCallback->Initialize(m_pWrappedProfilerInfo);
-    }
-    else
-    {
-        hr = pCorProfilerCallback->Initialize(m_pRealProfilerInfo);
-    }
-
```

Comment:
You may want to maintain the error semantics that we have now. The other Initialize call doesn't care about the HRESULT of the raw profiler hook. This one does.

Reply:
Good catch!

---
In reply to: [419803414](https://github.com/microsoft/CLRInstrumentationEngine/pull/256#discussion_r419803414) [](ancestors = 419803414)